File: Source/Entities/Crossover/Map/MapController.cs
using Celeste.Mod.Entities;
using Microsoft.Xna.Framework;
using Monocle;

namespace Celeste.Mod.KoseiHelper.Entities.Crossover;

[CustomEntity("KoseiHelper/MapController")]
[Tracked]
public class MapController : Entity
{
    public string pinMadeline, pinBerry;
    public string[] mapLayers;
    public string rootPath;
    public bool onlyOnSafeGround;
    public string mapDisabledFlag;
    public MapController(EntityData data, Vector2 offset) : base(data.Position + offset)
    {
        base.Tag = Tags.Global;
        pinMadeline = data.Attr("pinMadeline", "KoseiHelper/map/pins/pin_SingleDash");
        pinBerry = data.Attr("pinBerry", "collectables/strawberry");
        mapLayers = data.Attr("mapLayers", "filenames,separated,by,commas").Split(',');
        rootPath = data.Attr("rootPath", "");
        onlyOnSafeGround = data.Bool("onlyOnSafeGround", false);
        mapDisabledFlag = data.Attr("mapDisabledFlag", "KoseiHelper_CantOpenMap");
    }
}

File: Loenn/entities/Crossover/MapController.lua
local MapController = {}

MapController.name = "KoseiHelper/MapController"
MapController.depth = -20000
MapController.texture = "objects/KoseiHelper/Crossover/Controllers/MapController"
MapController.justification = {0.0, 0.0}
MapController.placements = {
    name = "MapController",
    data = {
		pinMadeline = "KoseiHelper/map/pins/pin_SingleDash",
		pinBerry = "collectables/strawberry",
		mapLayers = "Base,TestA,TestB",
		rootPath = "KoseiHelper/map/layers/",
		onlyOnSafeGround = false,
		mapDisabledFlag = "KoseiHelper_CantOpenMap"
	}
}

MapController.fieldInformation = {
	pinMadeline = {
		options = {
			"KoseiHelper/map/pins/pin_SingleDash",
			"KoseiHelper/map/pins/pin_DoubleDash",
			"KoseiHelper/map/pins/pin_Black_SingleDash",
			"KoseiHelper/map/pins/pin_Black_DoubleDash",
		},
		editable = true
	}
}

return MapController